frontend/src/features/leases/detail/LeasePages/surplus/Surplus.test.tsx

```diff
@@ -0,0 +1,96 @@
+import { Formik } from 'formik';
+import { createMemoryHistory } from 'history';
+import { defaultFormLease, IFormLease, IProperty } from 'interfaces';
+import { noop } from 'lodash';
+import { mockOrganization, mockProperties, mockUser } from 'mocks/filterDataMock';
+import { prettyFormatDate } from 'utils';
+import { render, RenderOptions, RenderResult } from 'utils/test-utils';
+
+import Surplus from './Surplus';
+
+const history = createMemoryHistory();
+
+describe('Lease Surplus Declaration', () => {
+  const setup = (renderOptions: RenderOptions & { lease?: IFormLease } = {}): RenderResult => {
+    // render component under test
+    const result = render(
+      <Formik onSubmit={noop} initialValues={renderOptions.lease ?? defaultFormLease}>
+        <Surplus />
+      </Formik>,
+      {
+        ...renderOptions,
+        history,
+      },
+    );
+    return result;
+  };
+
+  it('renders as expected', () => {
+    const result = setup({
+      lease: { ...defaultFormLease, persons: [mockUser], organizations: [mockOrganization] },
+    });
+    expect(result.asFragment()).toMatchSnapshot();
+  });
+  it('Snapshot matches', () => {
+    const result = setup({
+      lease: { ...defaultFormLease, persons: [mockUser], organizations: [mockOrganization] },
+    });
+    expect(result.asFragment()).toMatchSnapshot();
+  });
+
+  it('Table row count is equal to the number of properties', () => {
+    const result = setup({
+      lease: {
+        ...defaultFormLease,
+        properties: mockProperties,
+        persons: [mockUser],
+        organizations: [mockOrganization],
+      },
+    });
+    const rows = result.container.querySelectorAll('[role="row"]');
```

Comment:
React Testing Library provides **getByRole**, **getAllByRole** that should be used whenever possible.
https://testing-library.com/docs/queries/byrole/

```tsx
const rows = result.getAllByRole('row'); 
...
```

Reply:
that actually sounds way clearer! Will update the tests

Reply:
updated the top level. The subquery was needed since it was selecting inside an html element